File: 10573-415-763-partition-labels/10573-415-763-partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> list[int]:
        lst = [0] * 26
        
        for i in range(len(s)):
            lst[ord(s[i]) - ord('a')] = i
        
        result = []
        start = 0
        end = 0
        
        for i in range(len(s)):
            end = max(end, lst[ord(s[i]) - ord('a')])
            
            if end == i:
                result.append(end - start + 1)
                start = end + 1
        
        return result

File: 10573-415-763-partition-labels/10573-415-763-partition-labels.js
/**
 * @param {string} s
 * @return {number[]}
 */
var partitionLabels = function(s) {
    const lastIndex = new Array(26).fill(0);
    
    for (let i = 0; i < s.length; i++) {
        lastIndex[s.charCodeAt(i) - 'a'.charCodeAt(0)] = i;
    }
    
    const result = [];
    let start = 0;
    let end = 0;
    
    for (let i = 0; i < s.length; i++) {
        end = Math.max(end, lastIndex[s.charCodeAt(i) - 'a'.charCodeAt(0)]);
        
        if (end === i) {
            result.push(end - start + 1);
            start = end + 1;
        }
    }
    
    return result;
};